### DISCUSSION TITLE
Group/Direct Association - Zooz Scene controller to Zooz switch - 2-way communication

### DISCUSSION THREAD

**Discussion by nateww** [zwave-js/zwave-js-ui#4054] [Q&A]
So, I'm trying to do direct association (Group in ZWaveJS-UI).

The two devices are:

- Zen32 - https://www.getzooz.com/zooz-zen32-scene-controller/
- Zen72 - https://www.getzooz.com/zooz-zen72-dimmer/

I am trying to setup the scene controller (based 1 big and 4 little buttons) such that I can control a set of lights/dimmer in my house remotely.

I am able to successfully setup the ability to control the Zen72 from the Zen32 using an association (yay!).
![image](https://github.com/user-attachments/assets/dd3bdc6d-2903-421a-851e-08413032eba7)

However, this doesn't work well when the switch changes the state, as I want to have the state pushed to the scene controller, so I can update the state of the LED indicating it's on/off.

I can setup a association from the Zen72 -> Zen32, but it only allows me to associate with the primary (big) button, there is no way (that I can find) to associate the switch with the alternate buttons.

If I'm reading the documetnation correctly, there should be a way to associate with the smaller switches using a different group.

https://www.support.getzooz.com/kb/article/608-zen32-scene-controller-advanced-settings/

![image](https://github.com/user-attachments/assets/69113fe7-728d-45de-a5a4-95029f158ff3)

I'm trying to figure out how to associate from the Zen72 -> Group4 on the little button, instead of associating with the main relay (Big button).

![image](https://github.com/user-attachments/assets/897cff7a-2554-403c-9b3d-096d789c2f36)

Is this simply not possible in that I can only do 1-way communication.

Thanks for any help you can provide!

If that doesn't work, do you have any idea how I can set things up to avoid having to write automations for this simple stuff.

On a completely different note, how do I setup scenes?  Everytime I add a number device, I get a crap-ton of number entities, as well as a bunch of other 'switch' and 'sensor' entities that APPEAR to be related to scenes.  But, I couldn't find any documentation describing how to set them up in ZWaveJS-UI.

Thanks in advance!

**Comment by kpine** [comment]
> However, this doesn't work well when the switch changes the state, as I want to have the state pushed to the scene controller, so I can update the state of the LED indicating it's on/off.

To sync the LED state you will need a hub (like Home Assistant), as the small button LEDs are controlled by either setting configuration parameters or using the indicator command class. Neither of those are something your light switch is capable of doing on its own by associations. See the Zooz [product FAQ](https://www.support.getzooz.com/kb/article/1418-zen32-scene-controller-faqs/):

> if you want to control the LED and keep it in sync with the device the button on the ZEN32 is controlling, you can program to control the LED in a separate rule or automation. That means that the LED indicators can be controlled as regular devices and used for notifications, completely separate of the buttons (**as long as the functionality is supported on your Z-Wave hub**).

and [HA programming how-to](https://www.support.getzooz.com/kb/article/601-how-to-program-your-zen32-scene-controller-on-home-assistant/):

>  the buttons on the ZEN32 don't report on/off status as they are not on/off devices themselves

They go on to explain how to do this in HA (if you are using it) by setting config parameters (but show node-red for some reason?). The linked video tutorial might have a better explanation. You should also be able to control them easier in HA using the "indicator" entities (`switch` entities I believe). Basically: watch Zen72 for state changes in automation software, push setting or set indicator based on state to zen32.

> I can setup a association from the Zen72 -> Zen32, but it only allows me to associate with the primary (big) button, there is no way (that I can find) to associate the switch with the alternate buttons.
> 
> If I'm reading the documetnation correctly, there should be a way to associate with the smaller switches using a different group.

Groups 4-11 of the Zen32 allow controlling other devices with the small buttons. E.g. Group 4 configures small button 1 to turn on/off another switch, and Group 5 configures small button 1 to turn on/off/dim another switch, etc. They are not used for controlling the Zen32 by other devices.

> I'm trying to figure out how to associate from the Zen72 -> Group4 on the little button, instead of associating with the main relay (Big button).
>
> Is this simply not possible in that I can only do 1-way communication.

Correct, groups are 1-way communication. Associations are always for controlling other devices. You don't configure a remote device to communicate with "Group 4". 

> On a completely different note, how do I setup scenes? Everytime I add a number device, I get a crap-ton of number entities, as well as a bunch of other 'switch' and 'sensor' entities that APPEAR to be related to scenes. But, I couldn't find any documentation describing how to set them up in ZWaveJS-UI.

Since it sounds like you're using Home Assistant, by the mention of "entities", you can just program the device using HA's scenes or scripts/automations. Scenes in Z-Wave JS UI are for use with with the MQTT API: https://zwave-js.github.io/zwave-js-ui/#/guide/mqtt?id=execute-scene, and you can just ignore them, unless of course you are using MQTT directly instead of the Z-Wave integration.

**Comment by nateww** [comment]
Thanks @kpine.  I really appreciate the prompt response.

    Groups 4-11 of the Zen32 allow controlling other devices with the small buttons. E.g. Group 4 configures small button 1 to 
    turn on/off another switch, and Group 5 configures small button 1 to turn on/off/dim another switch, etc. They are not used 
    for controlling the Zen32 by other devices.

Dang, I was hoping that wasn't the case, but given my lack of understanding of the ZWave protocol and ZWaveJS internals, it's not a total shocker.

And, this lack of understanding didn't allow me to realize when reading the product FAQ that I couldn't do direct two-way associations.

Also, my particular situation is somewhat 'special' (not necessarily in a good way) in that my HA instance is located 100 miles away from the location of my ZWave devices.  My remote home is connected over a VPN using a slow and mostly usable network connection, and over the last 7+ years I've found that using a local MQTT broker running in bridged mode (with a few QOS tweaks) allows the remote site to be monitored/controlled effectively. Unfortunately, other communications protocols haven't worked as well (such as using the websocket from ZWaveJS <-> HA).

Historically, for monitoring systems that send out occasional notifications (temperature settings, etc..), the slow response time of MQTT over the link is more than adequate.  Even if someone triggers the motion or door sensors, the time taken to receive and respond to the event is minuscule compared the time it takes to deliver the notification over email and/or IM to my phone. And, if my temperature setting comes in late 5-10 seconds late, it won't make any difference in my quality of life.

The last few months I've been upping my automation game lately, slowly replacing all the dumb light switches with ZWave switches, which is very much showing my lack of knowledge. With the new switches and the use of direct associations for some of the 3-way setups, the response time has been fantastic.  And because the switches are directly connected to the lights, my solutions so far have high spousal-approval factor.  However, my fear is that relying on automation which involve send messages 100 miles away (round-trip times can be up to 500ms for a complete request/response), will be annoying to the spousal unit due to it not reacting quickly, meaning my automation dreams will be crushed. :)

Due to latency issues, and occasional unreliability of the communications link, I'm currently stuck with using MQTT only.  So, the HA programming/setup with **bulk_set_partial_config_parameters** won't work for me as it relies on the websocket between HA and ZWaveJS.

I'll play around with the [MQTT API:](https://zwave-js.github.io/zwave-js-ui/#/guide/mqtt?id=execute-scene) from above, and see if I can create something that runs on the locally hosted NUC at the remote site that hosts ZWaveJS and the MQTT broker.  I could probably build something quickly in Python or even a simple bash script that listens for MQTT events that pushes the updates to the Zen32 controller, thus avoiding the communications delays to/from HA. But, I need to familiarize myself with the MQTT setup first.

Is there anything I need to do to enable the MQTT API, as I've not seen any messages in my MQTT broker on those topics (specifically the _EVENTS_ topic)?

*<mqtt_prefix>/_EVENTS_/ZWAVE_GATEWAY-<mqtt_name>/<driver|node|controller>/<event_name>*

If there are any other documentations (blog articles, example code, etc..) that you could provide that would be useful to help me get started, I would greatly appreciate it.  The API documentation mentions the _EVENTS_ topic, but most of the examples mention a different path, like *zwavejs/_CLIENTS/ZWAVE_GATEWAY-<yourName>/api/sendCommand/set* (where I'm assuming *zwavejs* is equivalent to *<mqtt_prefix>*.  Once I get this figured, perhaps it would be useful to create a PR/MR to change the documentation to be more consistent with terms, but I'm not yet comfortable enough with the API to know what I'm doing.

Again, thank you for your response and answer.
